assets/js/components/ExecutionResults/ExecutionResults.jsx

```diff
@@ -15,65 +14,101 @@ import {
   ResultsContainer,
   HostResultsWrapper,
   CheckResult,
-  getHosts,
   getChecks,
   getHealth,
   getCheckResults,
   getCheckDescription,
 } from '@components/ChecksResults';
 import { UNKNOWN_PROVIDER } from '@components/ClusterDetails/ClusterSettings';
+import NotificationBox from '@components/NotificationBox';
+
+import { updateLastExecution } from '@state/actions/lastExecutions';
+import { getLastExecution } from '@state/selectors/lastExecutions';
+import { updateCatalog } from '@state/actions/catalog';
+import { getCatalog } from '@state/selectors/catalog';
 
 const truncatedClusterNameClasses =
   'font-bold truncate w-60 inline-block align-top';
 
-const getLabel = (health, expectations, failedExpectations) =>
-  health === 'passing'
-    ? `${expectations}/${expectations} expectations passed`
-    : `${failedExpectations}/${expectations} failed`;
+const getLabel = (status, health, error, expectations, failedExpectations) => {
+  if (status === 'running') {
+    return '';
+  }
+
+  if (error) {
+    return error;
+  }
+
+  if (health === 'passing') {
+    return `${expectations}/${expectations} expectations passed`;
+  }
+
+  return `${failedExpectations}/${expectations} expectations failed`;
+};
 
 function ExecutionResults({
   clusterID,
-  executionID,
   clusterName,
   cloudProvider,
   hostnames = [],
-  onExecutionFetch = getExecutionResult,
-  onCatalogFetch = getCatalog,
   onCatalogRefresh = () => {},
 }) {
-  const [loading, setLoading] = useState(false);
-  const [executionData, setExecutionData] = useState(null);
-  const [catalog, setCatalog] = useState(null);
   const [selectedCheck, setSelectedCheck] = useState(null);
   const [modalOpen, setModalOpen] = useState(false);
 
+  const dispatch = useDispatch();
+
+  const {
+    loading: catalogLoading,
+    data: catalog,
+    error: catalogError,
+  } = useSelector(getCatalog());
+
+  const lastExecution = useSelector(getLastExecution(clusterID));
+
   useEffect(() => {
-    setLoading(true);
-    Promise.all([onExecutionFetch(executionID), onCatalogFetch()])
-      .then(
-        ([{ data: fetchedExecutionData }, { data: fetchedCatalogData }]) => {
-          setLoading(false);
-          setExecutionData(fetchedExecutionData);
-          setCatalog(fetchedCatalogData.items);
-        }
-      )
-      .catch((error) => {
-        setLoading(false);
-        logError(error);
-      });
-  }, [onExecutionFetch, onCatalogFetch, setExecutionData, setCatalog]);
-
-  if (loading) {
+    dispatch(updateCatalog());
+    dispatch(updateLastExecution(clusterID));
```

Comment:
Could we pass these as props? Also, `dispatch(updateCatalog())` should be already available as `onCatalogRefresh`